examples/interfaces/http.py

```diff
@@ -0,0 +1,83 @@
+import json
+from collections import namedtuple
+
+from juju.interface import InterfaceBase
+
+
+class HTTPInterfaceProvides(InterfaceBase):
+    """Provides side for the 'http' interface protocol.
+
+    Example usage:
+
+        class MyCharm(CharmBase):
+            website = Endpoint(HTTPInterfaceProvides)
+
+            def __init__(self, parent, key):
+                super().__init__(parent, key)
+                self.framework.observe(self.website.on.joined, self.on_website_joined)
+
+        def on_website_joined(self, event):
+            self.config = self.framework.model.config
+            for client in self.website.clients:
+                client.serve(hosts=[client.ingress_address],
+                             port=self.config['http_port'])
```

Comment:
If it serves, it is a server and not a client, isn't it? Might explain the TODO confusion below re: ingress_address vs binding_address.

Reply:
Hrm, the intention there was "serve this client" but yeah, it does seem confusing.  I'm not sure what better syntax to use, though.  `client.provide()`?  `client.publish_to()`?  `self.website.serve(client, host, port)`?